DSharpPlus/Clients/DiscordClient.cs

```diff
@@ -889,7 +901,9 @@ private async IAsyncEnumerable<DiscordGuild> GetGuildsInternalAsync(int limit =
         int lastCount;
         do
         {
-            int fetchSize = remaining > 100 ? 100 : remaining;
+            if (cancellationToken.IsCancellationRequested) yield break;
+            
+            int fetchSize = remaining > 200 ? 200 : remaining;
             IReadOnlyList<DiscordGuild> fetchedGuilds = await this.ApiClient.GetGuildsAsync( fetchSize, isbefore ? last ?? before : null, !isbefore ? last ?? after : null, withCount);
```

Comment:
Do we not support passing a cancellation token here?

Reply:
Wdym?

Reply:
Github at it's finest; the api call

Reply:
Our restclient doesnt support cancellations at all.